src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -1085,36 +1101,49 @@ public String constructManifestAndUserData(boolean graphAlreadyBuilt, boolean dr
         }
-        String otpRunnerManifestS3FilePath = joinToS3FolderURI(otpRunnerManifestS3Filename);
-        String otpRunnerManifestFileOnInstance = String.format(
-            "/var/%s/otp-runner-manifest.json",
-            getTripPlannerString()
-        );
 
         // upload otp-runner manifest to s3
         try {
             ObjectMapper mapper = new ObjectMapper();
             mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
-            if (!uploadStringToS3File(otpRunnerManifestS3Filename, mapper.writeValueAsString(manifest), dryRun)) {
+            if (
+                !uploadStringToS3File(
+                    getOtpRunnerManifestS3Filename(graphAlreadyBuilt),
+                    mapper.writeValueAsString(manifest),
+                    dryRun
+                )
+            ) {
                 return null;
             }
         } catch (JsonProcessingException e) {
             status.fail("Failed to create manifest for otp-runner!", e);
             return null;
         }
+        return manifest;
+    }
+
+    /**
+     * Construct the user data script (as string) that should be provided to the AMI and executed upon EC2 instance
+     * startup.
+     */
+    public String constructUserData(boolean graphAlreadyBuilt) {
+        String otpRunnerManifestS3FilePath = joinToS3FolderURI(getOtpRunnerManifestS3Filename(graphAlreadyBuilt));
+        String otpRunnerManifestFileOnInstance = String.format(
+            "/var/%s/otp-runner-manifest.json",
+            getTripPlannerString()
+        );
 
-        //////////////// BEGIN USER DATA
         List<String> lines = new ArrayList<>();
         lines.add("#!/bin/bash");
         // NOTE: user data output is logged to `/var/log/cloud-init-output.log` automatically with ec2 instances
         // Add some items to the $PATH as the $PATH with user-data scripts differs from the ssh $PATH.
         lines.add("export PATH=\"$PATH:/home/ubuntu/.yarn/bin\"");
         lines.add(String.format("export PATH=\"$PATH:/home/ubuntu/.nvm/versions/node/%s/bin\"", NODE_VERSION));
         // Remove previous files that might have been created during an Image creation
-        lines.add(String.format("rm %s/%s || echo '' > /dev/null", webDir, OTP_RUNNER_STATUS_FILE));
+        lines.add(String.format("rm %s/%s || echo '' > /dev/null", EC2_WEB_DIR, OTP_RUNNER_STATUS_FILE));
         lines.add(String.format("rm %s || echo '' > /dev/null", otpRunnerManifestFileOnInstance));
-        lines.add(String.format("rm %s || echo '' > /dev/null", manifest.jarFile));
-        lines.add(String.format("rm %s || echo '' > /dev/null", manifest.otpRunnerLogFile));
+        lines.add(String.format("rm %s || echo '' > /dev/null", getJarFileOnInstance()));
+        lines.add(String.format("rm %s || echo '' > /dev/null", OTP_RUNNER_LOG_FILE));
         lines.add("rm /var/log/otp-build.log || echo '' > /dev/null");
         lines.add("rm /var/log/otp-server.log || echo '' > /dev/null");
```

Comment:
Should we just turn this into an array of fileNames with a forEach? E.g.:
```java
        // Remove previous files that might have been created during an Image creation
        String statusFile = String.join("/", EC2_WEB_DIR, OTP_RUNNER_STATUS_FILE);
        List<String> filesToRemove = Arrays.asList(statusFile, otpRunnerManifestFileOnInstance, getJarFileOnInstance(), OTP_RUNNER_LOG_FILE, "/var/log/otp-build.log", "/var/log/otp-server.log");
        filesToRemove.forEach(file -> lines.add(String.format("rm %s || echo '' > /dev/null", file)));
```

Reply:
Changes in #364.

Reply:
proposed changes accepted